test/cql-pytest/test_guardrail_replication_strategy.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+from util import unique_name, config_value_context
+from cassandra.protocol import ConfigurationException
+
+def create_ks_and_assert_warnings_count(cql, ks_opts, warnings_count = 0, fails_count = 0):
+    keyspace = unique_name()
+    if fails_count:
+        with pytest.raises(ConfigurationException):
+            cql.execute_async("CREATE KEYSPACE " + keyspace + " " + ks_opts).result()
+        return
+    else:
+        response_future = cql.execute_async("CREATE KEYSPACE " + keyspace + " " + ks_opts)
+        response_future.result()
+        cql.execute("DROP KEYSPACE " + keyspace)
+        if warnings_count:
+            assert response_future.warnings is not None and len(response_future.warnings) == warnings_count
+        else:
+            assert response_future.warnings is None
+
+def clear_guardrails_config(cql):
+    cql.execute(f"UPDATE system.config SET value = '' where name = 'replication_strategy_warn_list'")
+    cql.execute(f"UPDATE system.config SET value = '' where name = 'replication_strategy_fail_list'")
+
+def test_replication_strategy_warn_list(cql, this_dc):
```

Comment:
A test that by default creating/altering a ks and using SimpleStrategy generates a warning is missing and has to be added.

Reply:
Yes, exactly, a warning for SimpleStrategy is embedded in the config.cc, and only repeated - and commented out - in scylla.yaml, which just shows what the source code default is.